cmds/contest/main.go

```diff
@@ -82,9 +107,37 @@ var userFunctions = map[string]interface{}{
 	},
 }
 
+func expandArgument(arg string) string {
+	// it does not support correct expanding into depth more one.
+	flag.CommandLine.VisitAll(func(f *flag.Flag) {
+		arg = strings.Replace(arg, `%`+f.Name+`%`, f.Value.String(), -1)
```

Comment:
> I was wondering if instead we should give the plugins a way to register flags (in separate namespaces), and then give the possibility to those plugins to consume those flags at creation time.

Yeah, indeed it would be more structured/modular/flexible. I would like to do it. You would like to see it as a part of this PR of a separate one?

Reply:
> Can you should how this would look like with multiple arguments? This seems to be fragile ...

I do not see why is this fragile [I'm not telling it's not fragile, I just miss something :) ]. It does not support of expanding of variable to a value which is also required to be expanded, but if we will meet such case we can thing this through as well.

I did-not-yet think about deep expanding because of problems like this:
```
a = `%c%`
b = `%a%`
c = `%b%`
```

Reply:
If we go down the per plugin flags path, then definitely in a different PR.

Please, add some comment somewhere in the flag definition or here, to explain how passing arguments would look like. Maybe I missed it, but the comment for the flags says:

```
The engine to lock targets. Possible engines (the part before the first colon): %s
```

What comes after the `:`? Does it support multiple `%flag%` if separated by additional semicolons? Where are the flags that are replaced in the placeholder defined? Please help whoever is reading the code understand with as little barrier as possible. I don't think it's reasonable for me to have too look at `parseFactoryInfo`, `expandArgument`, `setupFlags` to understand how `flagTargetLocker works.

Reply:
Yeah, I right it's really confusing, thank you. I'll fix it :)

Reply:
Added more description. Please let me know if now it's clear :)